resource-service/src/component-tests/java/uk/gov/gchq/palisade/component/resource/service/ResourceServicePersistenceProxyTest.java

```diff
@@ -0,0 +1,140 @@
+    private ReactivePersistenceLayer persistenceLayer;
+    @Autowired
+    private Materializer materializer;
+
+    private final Function<Integer, ResourceRequest> requestFactoryObj = i -> ResourceRequest.Builder.create()
+            .withUserId("user-id")
+            .withResourceId(String.format("file:/test/resourceId/data%d.txt", i))
+            .withContext(new Context().purpose("test-purpose"))
+            .withUser(new User().userId("test-user"));
+
+    @BeforeEach
+    void setup() throws ExecutionException, InterruptedException {
+        Source.single(FILE_1)
+                .via(persistenceLayer.withPersistenceById(FILE_1.getId()))
+                .via(persistenceLayer.withPersistenceByType(FILE_1.getType()))
+                .via(persistenceLayer.withPersistenceBySerialisedFormat(FILE_1.getSerialisedFormat()))
+                .runWith(Sink.seq(), materializer);
+
+        TimeUnit.SECONDS.sleep(1);
+    }
+
+    @Test
+    void testContextLoads() {
+        assertThat(resourceServiceAsyncProxy).isNotNull();
+        assertThat(persistenceLayer).isNotNull();
+    }
+
+    @Test
+    void testGetResourcesByIdSuccess() throws ExecutionException, InterruptedException {
+        // Given a user request
+        final ResourceRequest resourceRequest = requestFactoryObj.apply(1);
+
+        // Then retrieving a resource from the cache
+        final Source<AuditableResourceResponse, NotUsed> subject = this.resourceServiceAsyncProxy.getResourcesById(resourceRequest);
+        final CompletionStage<List<AuditableResourceResponse>> future = subject.runWith(Sink.seq(), materializer);
+        final List<AuditableResourceResponse> result = future.toCompletableFuture().get();
+
+        // Then check there is no error and check the returned resource ID
+        assertAll(
+                () -> assertThat(result).hasSize(1),
+                () -> assertThat(result.get(0).getAuditErrorMessage()).isNull(),
+                () -> assertThat(result.get(0).getResourceResponse().getResource().getId()).isEqualTo(FILE_1.getId())
+        );
+    }
+
+    @Test
+    void testGetResourceByIdFailure() {
+        // Given a user request
+        final ResourceRequest resourceRequest = requestFactoryObj.apply(2);
+
+        // Then retrieving a different resource from the cache
+        final List<AuditableResourceResponse> result = this.resourceServiceAsyncProxy.getResourcesById(resourceRequest)
+                .runWith(Sink.seq(), materializer)
+                .toCompletableFuture().join();
+
+        // Then check there is an error
+        assertAll(
+                () -> assertThat(result).hasSize(1),
+                () -> assertThat(result.get(0).getResourceResponse()).isNull(),
+                () -> assertThat(result.get(0).getAuditErrorMessage()).isNotNull(),
+                () -> assertThat(result.get(0).getAuditErrorMessage().getError().getCause().getMessage())
+                        .startsWith("Failed to walk path")
```

Comment:
If you're doing an assertion on the error you dont need to do a null check. 
the below assertion is a bit more of a relaible way to assert exceptions
```suggestion
                () -> assertThat(result.get(0).getAuditErrorMessage().getError())
                        .isExactlyInstanceOf(NoSuchResourceException.class)
                        .getCause()
                        .hasMessageContaining("Failed to walk path \\test\\resourceId\\data2.txt")
```

Reply:
Updated